tests/integration_test.rs

```diff
@@ -0,0 +1,133 @@
+extern crate addr2line;
+extern crate itertools;
+extern crate glob;
+
+use std::env;
+use std::path;
+use std::process;
+use std::io::prelude::*;
+use itertools::Itertools;
+
+#[test]
+fn identity_map() {
+    let target = env::current_exe().unwrap();
+    let mut debug = target.clone();
+
+    // point us at the real macOS debugging files
+    if cfg!(target_os = "macos") {
+        let fname = debug.file_name().unwrap().to_string_lossy().into_owned();
+        debug.set_file_name(format!("{}.dSYM", fname));
+        debug.push("Contents");
+        debug.push("Resources");
+        debug.push("DWARF");
+        debug.push(fname);
+    }
+
+    // create our mapping
+    let ours = addr2line::Mapping::new(&debug).unwrap();
+
+    // spin up the "real" addr2line
+    let (theirs, exe) = if cfg!(target_os = "macos") {
+        ("atos", "-o")
+    } else {
+        ("addr2line", "-e")
+    };
+    let mut theirs = process::Command::new(theirs)
+        .arg(exe)
+        .arg(&target)
+        .stdin(process::Stdio::piped())
+        .stdout(process::Stdio::piped())
+        .spawn()
+        .expect("failed to spawn their addr2line");
+
+    // find a bunch of addresses
+    let names = process::Command::new("/usr/bin/nm")
+        .arg(&target)
+        .output()
+        .expect("failed to execute nm");
```

Comment:
Can you combine the `t` filtering and every fifth address filtering with this `nm`ing into a function named `get_test_addresses`? That will really improve the readability for me here, and clearly separate gathering test data from feeding it into the `addr2line` executables.